pyfr/plugins/integrate.py

```diff
@@ -172,6 +165,29 @@ def _eval_exprs(self, intg):
 
         return intvals
 
+    def _grad_pvars(self, intg):
+        # Fetch the corrected gradients
+        grad_soln = intg.grad_soln
+
+        grads_eles = []
+        # Iterate over each element type in the simulation
+        for i, (soln, eleinfo) in enumerate(zip(intg.soln, self.eleinfo)):
```

Comment:
It's also worth noting that the integrate and time averaging plugins use different region/subsetting logic and so I am unsure how much code they can share.  (Integrate can sometimes select specific points from specific elements, whereas tavg will also work in terms of full elements.)

Reply:
Yes, the last version includes `_grad_pvars` with the logic of `tavg`

Reply:
Okay, would you be able to test if everything still works as expected for regions which 'cut' through an element.  (Here PyFR should just include the quadrature points which are inside the region for boundary elements; which is a reasonable-ish assumption.)

Reply:
There was a bug when meshes had more than one type of element due to the fact that `grad_soln` was being reused when calling `np.rollaxis`. This has been solved in the last commit.

I have tried the inc-cylinder-2d test case and it seems that `tavg` works okeysh even when using an arbitrary region box, I don't know if you have a better validation testcase for the `tavg` plugin. I'll leave attached a picture of the mean `u` velocity in the cylinder test case (the averaging starts from `t=0` so this result is not perfect but it should validate a bit the implementation)

![image](https://user-images.githubusercontent.com/11050889/107158625-bb610380-698b-11eb-8ac7-8caa0d8e8268.png)

Reply:
I'll try tomorrow to output the mean velocity gradients

Reply:
The other bug I watch out for is when a region only has a single type of element (but the partition itself has two or more element types).